src/ActionInputValidator/Validations.ts

```diff
@@ -28,7 +28,8 @@ export function validateAppDetails() {
 
     if(!!actionParams.appName || (!!actionParams.slotName && actionParams.slotName.toLowerCase() !== 'production')) {
         let creds: ScmCredentials = PublishProfile.getPublishProfile(actionParams.publishProfileContent).creds;
-        let splitUsername: string[] = creds.username.toUpperCase().substring(1).split("__");
+        //for kubeapps in publishsettings file username doesn't start with $, for all other apps it starts with $

```

Comment:
Lib file corresponding to this file is missing.

Reply:
added in new commit